[MemoryViz] Improve description of blocks with missing frames

aaronenyeshi opened the pull request:
Summary:
It is common for blocks to be missing frames and there are many users asking why.

Let's improve this output message to cover common reasons:

1) block was allocated before _record_memory_history was enabled
2) context or stacks passed to _record_memory_history does not include this block
3) backward events allocated with C++ stack and will not show if stacks = python

Test Plan:
CI and ran it locally:
![image](https://github.com/pytorch/pytorch/assets/17602366/60a03a22-0e3e-43d8-9ee7-b14358096fc7)

Differential Revision: D56490921

Pulled By: aaronenyeshi

aaronenyeshi commented:
@pytorchbot rebase

zdevito commented on torch/utils/viz/MemoryViz.js:
```diff
@@ -762,7 +762,12 @@ function frameFilter({name, filename}) {
+      `<block has no frames, potential causes:\n` +
+      `1) block was allocated before _record_memory_history was enabled,\n` +
+      `2) context or stacks passed to _record_memory_history does not include this block,\n` +
+      `3) backward events allocated with C++ stack and will not show if stacks = python>`
```
1 is actionable, 3 can be rephrased to be actionable. Can 2 be reworded to describe how to change _record_memory_history in order to get information?

```
3) This event occurred during backward, which has no python frames, and memory history did not include C++ frames. Use stacks='all' to record both C++ and python frames.
```

aaronenyeshi replied:
Sounds good, I've updated the string, and matched similar style (capitalization, periods, well-formed sentences).

zdevito reviewed (approved):
Thanks for making this more clear!

aaronenyeshi commented:
New picture after addressing review:
![image](https://github.com/pytorch/pytorch/assets/17602366/b2df16b0-7735-474f-ad82-9f06f452b8d4)

aaronenyeshi committed:
[MemoryViz] Improve description of blocks with missing frames (#124784)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/124784

It is common for blocks to be missing frames and there are many users asking why.

Let's improve this output message to cover common reasons:

1) block was allocated before _record_memory_history was enabled
2) context or stacks passed to _record_memory_history does not include this block
3) backward events allocated with C++ stack and will not show if stacks = python

Test Plan:
CI and ran it locally:
 {F1495469480}

Differential Revision: D56490921

Pulled By: aaronenyeshi

aaronenyeshi commented:
@pytorchbot merge
